### PR TITLE
rdpsnd/winmm: fix SetVolume when device is not opened.

### PR DESCRIPTION
**Pull request by llyzs**
The client may receive SetVolume PDU before receiving the first audio sample (which triggers the opening of the audio device in winmm plugin). This caused the volume PDU be lost and after some recent changes, now causes the client to be closed unexpectedly. Thanks to the error checking changes so that we can detect this problem.

### CODE REVIEW

**Comment by llyzs** [comment]
@akallabeth @mfleisz

**Comment by akallabeth** [comment]
+1

File: channels/rdpsnd/client/winmm/rdpsnd_winmm.c [hunk 1]
```diff
@@ -375,6 +386,8 @@ UINT freerdp_rdpsnd_client_subsystem_entry(PFREERDP_RDPSND_DEVICE_ENTRY_POINTS p
 		return CHANNEL_RC_NO_MEMORY;
 	}
 
+	winmm->volume = 0xFFFFFFFF;
+
```
Comment (hardening) [inline channels/rdpsnd/client/winmm/rdpsnd_winmm.c]: Just wondering if it's a good idea to start at full volume...

Comment (llyzs) [inline channels/rdpsnd/client/winmm/rdpsnd_winmm.c]: @hardening 

This is the application volume, not master volume. In fact OS default of the application volume is 100% so this should not be a problem.

Comment (hardening) [inline channels/rdpsnd/client/winmm/rdpsnd_winmm.c]: Nice then, thank you for the answer.

**Pull request merged**